src/mlpack/tests/softmax_regression_test.cpp

```diff
@@ -362,12 +362,8 @@ BOOST_AUTO_TEST_CASE(SoftmaxRegressionTrainTest)
 
   // This should be the same as the default parameters given by
   // SoftmaxRegression.
-  SoftmaxRegressionFunction srf(dataset, labels, 2, 0.0001, false);
-  L_BFGS<SoftmaxRegressionFunction> lbfgs(srf);
-  SoftmaxRegression<> sr(lbfgs);
-
+  SoftmaxRegression<> sr(dataset, labels, 2, 0.0001, false);
   SoftmaxRegression<> sr2(dataset.n_rows, 2);
-  sr2.Parameters() = srf.GetInitialPoint(); // Start from the same place.
```

Comment:
The idea here was just to make sure that the optimizations start from the same initial point, but we have to be a little more tricky now about how to do that since we can't get the initial point as easily---we have to use the constructor that initializes the parameters but does not train:

```
SoftmaxRegression<> sr(dataset.n_rows, 2);
SoftmaxRegression<> sr2(dataset.n_rows, 2);

// Force the same initial optimization point.
sr.Parameters() = sr2.Parameters();

// Now train both models.
sr.Train(dataset, labels, 2);
sr2.Train(dataset, labels, 2);
```